### DISCUSSION TITLE
Graphql playground does not update browser url

### DISCUSSION THREAD

**Discussion by thanhpd56** [99designs/gqlgen#2044] [Ideas]
Hi, I'm using latest version of gqlgen. When I try to create new query or mutation on playground site, the url has not been updated.

I see that the graphiql support generate url base on our query or mutation.
For example: https://graphql.org/swapi-graphql?query=%7B%0A%09film(id%3A%201)%7B%0A%20%20%20%20title%0A%20%20%7D%0A%7D
Whenever I change the query, the url will be changed.

This feature will help us to share the query easily.
Question:
Does gqlgen support this feature?
If yes, how can I enable it?

Thanks.

**Comment by frederikhors** [comment]
Although it is true that the current graphiql does not change the URL I move this in requests. In the issues we are trying to leave only very urgent things to fix.

**Reply by thanhpd56** [reply]
Thanks.

**Comment by blakebutcher** [comment]
Being able to share queries with playground URLs would be very good.

https://github.com/graphql/graphiql/discussions/3450 has more information about how this can be done.

**Reply by blakebutcher** [reply]
gqlgen's bundled GraphiQL handler currently doesn't provide a way to initialise or persist query state via URL parameters. GraphiQL itself supports this pattern and includes example implementations that read query, variables, headers and operationName from the URL and keep them synchronised as the user edits. Could gqlgen expose this as an optional feature of the built-in GraphiQL handler so users can share GraphiQL URLs without maintaining a custom GraphiQL frontend?

Example implementation: https://github.com/graphql/graphiql/blob/1f9335051fffc9e6a6f950b6f8060ed521b56789/packages/graphiql/resources/renderExample.js

**Comment by StevenACoffman** [comment]
Sure! Can you make a PR? It looks like you would plop that in here: https://github.com/99designs/gqlgen/blob/master/graphql/playground/playground.go

**Reply by blakebutcher** [reply]
I hope I did this right https://github.com/99designs/gqlgen/pull/4311
